pkg/reportingplugins/mercury/v2/mercury.go

```diff
@@ -188,12 +193,17 @@ func (rp *reportingPlugin) Observation(ctx context.Context, repts ocrtypes.Repor
 	} else if obs.NativePrice.Val.Cmp(MissingPrice) <= 0 {
 		p.NativeFee = mercury.MaxInt192Enc
 	} else {
-		nativeFee := mercury.CalculateFee(obs.NativePrice.Val, rp.offchainConfig.BaseUSDFeeCents)
-		if nativeFeeEncoded, err := mercury.EncodeValueInt192(nativeFee); err != nil {
-			nativeErr = pkgerrors.Wrapf(err, "failed to encode native fee; val=%s", nativeFee)
+		nativeFee, err := mercury.CalculateFee(obs.NativePrice.Val, rp.offchainConfig.BaseUSDFeeCents)
```

Comment:
Is it possible for the price to be negative? For assets it's unusual, but if there is a bug where the NativePrice.Val is negative, CalculateFee would return negative, and this would be encoded to a uint192, given the most significant is the sign, this would in turn be an extraordinarily large fee and lead to unverifiable reports.

Reply:
Imo handling negative values doesn't belong to the fee calculation and should happen somewhere upstream. Also having unverifiable report in this case sounds like a good outcome to me. I'll let @samsondav to chime in here as the code owner

Reply:
If we have a bug in the software which leads to int_max fees being generated, there will be a complete outage. I think instead, we should remove the concept of int_max for failures and instead, zero the fee.

This also reduces gas costs quite significantly, as passing 0x00..00 is cheaper than passing 0x_int_max not only because 0 is billed for less, but low entropy data can easily be compressed when rolling up on a L2

Reply:
Core side will never encode a negative fee. But it doesn't hurt to have extra checks.